cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/LanguageFrontend.java

```diff
@@ -26,121 +26,50 @@
 package de.fraunhofer.aisec.cpg.frontends;
 
 import de.fraunhofer.aisec.cpg.TranslationConfiguration;
+import de.fraunhofer.aisec.cpg.graph.CodeAndLocationProvider;
+import de.fraunhofer.aisec.cpg.graph.LanguageProvider;
 import de.fraunhofer.aisec.cpg.graph.Node;
 import de.fraunhofer.aisec.cpg.graph.declarations.TranslationUnitDeclaration;
-import de.fraunhofer.aisec.cpg.graph.statements.GotoStatement;
-import de.fraunhofer.aisec.cpg.graph.statements.LabelStatement;
 import de.fraunhofer.aisec.cpg.passes.scopes.ScopeManager;
 import de.fraunhofer.aisec.cpg.sarif.PhysicalLocation;
 import de.fraunhofer.aisec.cpg.sarif.Region;
 import java.io.File;
 import java.util.*;
-import java.util.Map.Entry;
-import java.util.function.BiConsumer;
-import java.util.function.BiPredicate;
 import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public abstract class LanguageFrontend {
+public abstract class LanguageFrontend extends ProcessedListener
```

Comment:
This class should also have a quick javadoc explaining what it does and what it NOT does and how long the lifecycle of it is.

Reply:
While we are at it: Convert to Kotlin?